### ISSUE TITLE
Strange behaviour with text concealing

### ISSUE THREAD

**Issue by dbatten5** [preservim/vim-markdown#605]
I have the following settings set:
```
conceallevel=2
let g:vim_markdown_conceal=1
```

But for a markdown file with the following raw text:
```md
*test*
**test**
***test***
_test_
$x$
```

My output looks like this 
<img width="166" alt="image" src="https://user-images.githubusercontent.com/17271241/196001795-56d17a81-3b1a-4049-9f8c-64129d4e22c3.png">

Everything apart from the katex maths section is concealed and italics don't show as italiscised unless also set to bold (with `***`).  Is there some other setting I'm missing? Should I see the katex section properly concealed and the x replaced with an algebraic x, do I need to enable glyph font support or something for that?

**Comment by scanny** [comment]
Yeah, same here. Interestingly, a code fragment like `if x then a` enclosed in backticks _does_ appear in italics, so it's not that the terminal can't show them or anything. Just bold and italic, by themselves, don't appear in the alternate font.

**Comment by scanny** [comment]
Interesting, I didn't notice it until using a dark theme, but italic text is being rendered for me in the bold font and the italics don't appear until I use the `***` syntax at which point _both_ the bold and italic appear. For me this behavior is not sensitive to the `set conceallevel` or `g:vim_markdown_conceal` settings. It's also not sensitive to my colorscheme (same behavior using default).
